server/util/db/db.go

```diff
@@ -52,44 +53,53 @@ type DBHandle struct {
 	dialect       string
 }
 
-type Options interface {
-	ReadOnly() bool
-	AllowStaleReads() bool
-}
-
-type optionsImpl struct {
+type Options struct {
 	readOnly        bool
 	allowStaleReads bool
+	queryName       string
+}
+
+func Opts() *Options {
+	return &Options{}
 }
 
-func (oi *optionsImpl) ReadOnly() bool        { return oi.readOnly }
-func (oi *optionsImpl) AllowStaleReads() bool { return oi.allowStaleReads }
+func (o *Options) WithStaleReads() *Options {
+	o.readOnly = true
+	o.allowStaleReads = true
+	return o
+}
+
+// WithQueryName specifies the query label to use in exported metrics.
+func (o *Options) WithQueryName(queryName string) *Options {
+	o.queryName = queryName
+	return o
+}
 
 type DB = gorm.DB
 
 type txRunner func(tx *DB) error
 
-func StaleReadOptions() Options {
-	return &optionsImpl{
-		readOnly:        true,
-		allowStaleReads: true,
+func (dbh *DBHandle) gormHandleForOpts(ctx context.Context, opts *Options) *DB {
+	var db *DB
+	if opts.readOnly && opts.allowStaleReads && dbh.readReplicaDB != nil {
+		db = dbh.readReplicaDB
+	} else {
+		db = dbh.DB
+	}
```

Comment:
maybe default init db := dbh.DB and kill the else?

Reply:
Done.